tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,361 @@
+    if featureCheckResult["rc"] != 0:
+        pytest.skip("SONiC DUT isn't running supported version of SONiC")
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("sudo config portchannel retry-count set {} 5".format(port_channel))
+
+    # Wait for retry count info to be updated
+    time.sleep(5)
+
+    yield
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("sudo config portchannel retry-count set {} 3".format(port_channel))
+
+    # Wait for retry count info to be updated
+    time.sleep(90)
+
+
+@pytest.fixture(scope="function")
+def config_reload_on_cleanup(request, nbrhosts, duthost):
+    yield
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("sudo config reload -y")
+    config_reload(duthost, safe_reload=True)
+
+
+def log_lacpdu_packets(duthost, save_path):
+    """Capture LACPDU packets to file."""
+    # Support for LINUX_SLL2 was added in Scapy 2.5.0. We're using 2.4.5 currently.
+    start_pcap = "tcpdump -i any -y LINUX_SLL -w %s ether proto 0x8809" % (save_path)
+    stop_pcap = "sudo pkill -f '{}'".format(start_pcap)
+    start_pcap = "sudo nohup {} &".format(start_pcap)
+    duthost.shell(start_pcap)
+
+    time.sleep(30)
+
+    duthost.shell(stop_pcap, module_ignore_errors=True)
+
+
+@pytest.fixture(scope="function")
+def scapy_lacp_layer():
+    split_layers(scapy.contrib.lacp.SlowProtocol, scapy.contrib.lacp.LACP, subtype=1)
+    bind_layers(scapy.contrib.lacp.SlowProtocol, LACPRetryCount, subtype=1)
+    bind_layers(scapy.layers.l2.CookedLinux, scapy.contrib.lacp.SlowProtocol, proto=0x8809)
+
+    yield
+
+    split_layers(scapy.layers.l2.CookedLinux, scapy.contrib.lacp.SlowProtocol, proto=0x8809)
+    split_layers(scapy.contrib.lacp.SlowProtocol, LACPRetryCount, subtype=1)
+    bind_layers(scapy.contrib.lacp.SlowProtocol, scapy.contrib.lacp.LACP, subtype=1)
+
+
+def check_lacpdu_packet_version(duthost):
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+    port_channels = cfg_facts["PORTCHANNEL"].keys()
+    log_lacpdu_packets(duthost, "lacpdu.pcap")
+    lacpduPackets = None
+    with tempfile.NamedTemporaryFile() as tmp_pcap:
+        duthost.fetch(src="lacpdu.pcap", dest=tmp_pcap.name, flat=True)
+        lacpduPackets = rdpcap(tmp_pcap.name)
+    for port_channel in port_channels:
+        port_channel_status = duthost.get_port_channel_status(port_channel)
+        for port, status in list(port_channel_status["ports"].items()):
+            sendVersionVerified = False
+            receiveVersionVerified = False
+            actorInfo = status["runner"]["actor_lacpdu_info"]
+            partnerInfo = status["runner"]["partner_lacpdu_info"]
+            for pkt in lacpduPackets:
+                if pkt["LACPRetryCount"].version == 0xf1:
+                    if pkt["LACPRetryCount"].actor_system == actorInfo["system"] \
```

Comment:
I don't completely understand what's happening in below if and elif checks. Can you elaborate and add some info in comments for folks like me? :)